### DISCUSSION TITLE
device_get_binding() return NULL

### DISCUSSION THREAD

**Discussion by mejtejt** [zephyrproject-rtos/zephyr#29241] [Q&A] [answered]
Hi, 
This line of code   `dev = device_get_binding("BQ274XX");` always returns `NULL` even though I can clearly see in debugger that content is there :
![image](https://user-images.githubusercontent.com/33463717/96166911-eaf08980-0f1e-11eb-8397-47ae1422f901.png)

Does anyone have a clue why is this happening? This is a whole section of previous line and breakpoint is at `dev=...`: 

```
  dev = device_get_binding("BQ274XX");
  if (!dev) {
    std::cout << "Error while initializing BQ274XX" << std::endl;
  }
```
BQ274XX is configured in .dts file : 
![image](https://user-images.githubusercontent.com/33463717/96168065-833b3e00-0f20-11eb-97cb-2ccfe1c0528f.png)
Code is building fine with all additional config lines like CONFIG_I2C, CONFIG_I2C, CONFIG_SENSOR, CONFIG_BQ274XX in a prj.conf file.
Board I'm using is a prototype but with STM32L462VE MCU and mostly  of `<st/l4/stm32l462Xe.dtsi>` device tree config.
Not sure if I gave all the necessary info.
Thanks, cheers.

**Comment by mbolivar-nordic** [comment, marked as answer]
There are two reasons why device_get_binding() may return NULL:

1. There is no such device
2. The device's initialization function failed

Since from your report it looks like there is such a device, I assume its initialization function failed. You will need to add printlines or attach a debugger to that function and step through to figure out why.

**Reply by mejtejt** [reply]
It was device's initialization function failure. More precisely,  BQ27421 ID register couldn't be read because clock frequency was to fast. Had to switch it to `I2C_BITRATE_STANDARD`.
Thanks!
